pyproject.toml

```diff
@@ -17,18 +17,33 @@
 [build-system]
 build-backend = 'mesonpy'
 requires = [
-    "meson-python>=0.15.0",
-    "Cython>=3.0.8",        # when updating version, also update check in meson.build
-    "pybind11>=2.13.2",     # when updating version, also update check in scipy/meson.build
-    "pythran>=0.14.0",
+    # The upper bound on meson-python is pre-emptive only (looser
+    # on purpose, since chance of breakage in 0.18/0.19 is low with
+    # 0.17.1 working at time of writing)
+    "meson-python>=0.15.0,<0.20.0",
+    # we need at least Cython 3.1.0a1 for free-threaded CPython
+    # 3.13 wheel builds
+    "Cython>=3.1.0a1,<3.2.0; python_version=='3.13t'",        # when updating version, also update check in meson.build
```

Comment:
`3.13t` doesn't exist, so this is invalid, and there's no valid hook via https://packaging.python.org/en/latest/specifications/dependency-specifiers/#dependency-specifiers for accessing free-threaded status that I'm aware of...

Reply:
Correct. I suggest converting the "needs >=3.1.0a1 for free-threading" to a code comment here.

Reply:
> Correct. I suggest converting the "needs >=3.1.0a1 for free-threading" to a code comment here.

Not that it changes the lack of specifier for 3.13t, but some good news about cython 3.1: https://github.com/cython/cython/issues/6536

Reply:
There's only Discourse threads (IIRC there was another one). I don't feel like reopening that discussion though, since (a) free-threading as a non-default build is supposed to be a temporary state so no static metadata is defensible, and (b) there are way larger holes in specifiers, like not being able to distinguish between 32-bit and 64-bit Python.

Reply:
yay for Cython 3.1 soon 🤞🏼